### PR TITLE
Replace an `Assert` with `if` to enable constant values on INSERT..SELECT queries

### PR DESCRIPTION
**Pull request by onderkalaci**
This commit replaces an assert with an if to enable constant values on the
INSERT..SELECT queries.

### CODE REVIEW

**Comment by onderkalaci** [comment]
@sumedhpathak 

This is a very tiny bug fix only affecting a very specific part of  deparsing`INSERT .. SELECT` queries. Basically, it is replacing an assert with an if statement.

Given that we've not started release testing for `INSERT..SELECT` yet, the fix is very small, only affects a specific part of it and the PR includes a regression test.

I'm thinking that we should check-in this to master and 6.0 as well. The reason is that, this PR would allow users to specify constant values on the SELECT target list, which is useful to hard code roll-up times instead of using `now()`, which we don't support. Please see the regression test for the use-case.

Shall I go ahead and check this in?

**Comment by sumedhpathak** [comment]
@onderkalaci Is this a feature or a bug fix? I am not clear on why we had this Assert previously.
That said, the change seems small, so if you and Andres agree, and Andres can review/test the change, I am ok to add it to 6.0.

**Comment by jasonmp85** [comment]
Add the _backport_ label to keep track of this stuff (possibly even if only considering it?)!

It's nice to be able to filter all issues for "stuff we need to decide whether to backport" rather than needing context on every issue.

As this is for `INSERT...SELECT`, I'm assuming we don't need backport to 5.2, just 6.0, right?

File: src/test/regress/expected/multi_insert_select.out [hunk 1]
```diff
@@ -1176,6 +1176,69 @@ DEBUG:  sent COMMIT over connection 13300009
 DEBUG:  sent COMMIT over connection 13300009
 DEBUG:  sent COMMIT over connection 13300010
 DEBUG:  sent COMMIT over connection 13300010
+-- we don't want to see constraint vialotions, so truncate first
```
Comment (marcocitus) [inline src/test/regress/expected/multi_insert_select.out]: s/vialotions/violations/

File: src/test/regress/expected/multi_insert_select.out [hunk 2]
```diff
@@ -1176,6 +1176,69 @@ DEBUG:  sent COMMIT over connection 13300009
...
+DEBUG:  distributed statement: INSERT INTO public.raw_events_second_13300007 AS citus_table_alias (user_id, "time", value_1) SELECT user_id, 'Fri Oct 28 21:05:12.048697 2016'::timestamp without time zone AS "timestamp", value_1 FROM public.raw_events_first_13300003 raw_events_first WHERE ((hashint4(user_id) >= 1073741824) AND (hashint4(user_id) <= 2147483647)) RETURNING citus_table_alias.user_id, citus_table_alias."time"
+DEBUG:  ProcessQuery
+DEBUG:  Plan is router executable
+DEBUG:  CommitTransactionCommand
+DEBUG:  CommitTransaction
+DEBUG:  name: unnamed; blockState:       STARTED; state: INPROGR, xid/subid/cid: 0/1/0, nestlvl: 1, children: 
+DEBUG:  sent COMMIT over connection 13300007
+DEBUG:  sent COMMIT over connection 13300007
+DEBUG:  sent COMMIT over connection 13300005
+DEBUG:  sent COMMIT over connection 13300005
```
Comment (marcocitus) [inline src/test/regress/expected/multi_insert_select.out]: This output will change after a rebase and is likely to change again in the future, maybe avoid debugging output unless necessary.

**Comment by onderkalaci** [comment]
@anarazel @sumedhpathak 

Thinking/testing some more on this PR, I realize that the patch is not useful for the tests that are added (i.e., the added test already pass on the master branch as well).

**The story behind this PR:**
I hit this kind of a bug, which I don't remember how and I cannot replicate anymore. I had couple of different development branches for ```insert select```, which I suspect I had this bug in one of the development branches.

**Next Step:**

I'm thinking that we should close this PR and not merge to the master. I'll wait @anarazel to have a quick look on this in case he can think of any bugs related to that code path.

For adding the regression test, I'm already planning to check-in a PR in the couple of next weeks which increases the SQL coverage in the ```insert ... select``` tests. I can add this specific test in that PR as well.

**Comment by onderkalaci** [comment]
After chatting with andres, I'm closing this issue. Please see the above comment.

**Pull request closed without merging**